guide/ops/locations/_openstack.md

```diff
@@ -91,33 +143,95 @@ This is the same OpenStack location in a format that can be added to your
     brooklyn.location.named.My\ OpenStack.securityGroups=your-security-group
     brooklyn.location.named.My\ OpenStack.keyPair=your-keypair
 
-Chose a value of `your-flavor-id` from one of the defaults, or create your own flavor if
-you have administrator privileges.
-For for more information, see the
-[OpenStack flavors guide](http://docs.openstack.org/admin-guide/cli_manage_flavors.html).
 
-The default flavors are:
+### Troubleshooting
 
-    +-----+-----------+-----------+------+
-    | ID  | Name      | Memory_MB | Disk |
-    +-----+-----------+-----------+------+
-    | 1   | m1.tiny   | 512       | 1    |
-    | 2   | m1.small  | 2048      | 20   |
-    | 3   | m1.medium | 4096      | 40   |
-    | 4   | m1.large  | 8192      | 80   |
-    | 5   | m1.xlarge | 16384     | 160  |
-    +-----+-----------+-----------+------+
+#### Cloud Credentials Failing
 
-For an even more detailed example location configuration, consult the
-[template properties file](https://brooklyn.apache.org/v/latest/start/brooklyn.properties).
+If the cloud API calls return `401 Unauthorized` (e.g. in a `org.jclouds.rest.AuthorizationException`),
+then this could be because the credentials are incorrect.
 
+A good way to check this is to try the same credentials with the 
+[OpenStack nova command line client](http://docs.openstack.org/user-guide/common/cli_install_openstack_command_line_clients.html).
 
-### Other features
 
-Consult jclouds' [Nova template options](https://jclouds.apache.org/reference/javadoc/1.9.x/org/jclouds/openstack/nova/v2_0/compute/options/NovaTemplateOptions.html)
-for futher options when configuring Openstack locations.
+#### Unable to SSH: Wrong User
+
+If SSH authentication fails, it could be that the login user is incorrect. For most clouds, this 
```

Comment:
This problem can also manifest itself with an endless hang on the first SSH command after VMs have been provisioned -- I believe I've seen this when using OpenStack with the wrong username (e.g. "centos" instead of "ubuntu").